packages/rich-text-editor/src/_mixins.scss

```diff
@@ -14,8 +14,22 @@
     margin: 0 0 $spacing-sm;
   }
 
-  > p:last-child {
+  > *:last-child {
     margin-bottom: 0;
+    > li:last-of-type {
+      p {
+        margin-bottom: 0;
+      }
+    }
+  }
+
+  > *:first-child {
+    margin-top: 0;
+    > li:first-of-type {
+      p {
+        margin-top: 0;
+      }
+    }
   }
```

Comment:
for context to use the same selector without the wildcard it would look approximately like this:
```css
  > ul:last-child,
  > ol:last-child {
    margin-bottom: 0;
    > li:last-of-type {
      p {
        margin-bottom: 0;
      }
    }
  }

  > ul:first-child,
  > ol:first-child {
    margin-top: 0;
    > li:first-of-type {
      p {
        margin-top: 0;
      }
    }
  }
  
  ```